drivers/include/periph/i2c.h

```diff
@@ -0,0 +1,350 @@
+ */
+int i2c_init(i2c_t dev);
+
+/**
+ * @brief   Get mutually exclusive access to the given I2C bus
+ *
+ * In case the I2C device is busy, this function will block until the bus is
+ * free again.
+ *
+ * @param[in] dev           I2C device to access
+ *
+ * @return                  0 on success
+ */
+int i2c_acquire(i2c_t dev);
+
+/**
+ * @brief   Release the given I2C device to be used by others
+ *
+ * @param[in] dev           I2C device to release
+ */
+void i2c_release(i2c_t dev);
+
+/**
+ * @brief   Read data from the given I2C device
+ *
+ * @param[in]  dev          I2C peripheral device
+ * @param[in]  addr         7-bit or 10-bit device address (right-aligned)
+ * @param[out] data         memory location to store received data
+ * @param[in]  len          the number of bytes to read into @p data
+ * @param[in]  flags        optional flags (see @ref i2c_flags_t)
+ *
+ * @return                  I2C_ACK on successful transfer of @p len byte
+ * @return                  I2C_ADDR_NACK if response to address byte was NACK
+ * @return                  I2C_ERR for any other error
+ */
+int i2c_read(i2c_t dev, uint16_t addr, void *data, size_t len, uint8_t flags);
+
+/**
+ * @brief   Convenience function for reading data from a given register address
+ *
+ * @note    This function is using a repeated start sequence for reading from
+ *          the specified register address.
+ *
+ * @param[in]  dev          I2C peripheral device
+ * @param[in]  reg          register address to read from (8- or 16-bit,
+ *                          right-aligned)
+ * @param[in]  addr         7-bit or 10-bit device address (right-aligned)
+ * @param[out] data         memory location to store received data
+ * @param[in]  len          the number of bytes to read into @p data
+ * @param[in]  flags        optional flags (see @ref i2c_flags_t)
+ *
+ * @return                  I2C_ACK on successful transfer of @p len byte
+ * @return                  I2C_ADDR_NACK if response to address byte was NACK
+ * @return                  I2C_ERR for any other error
+ */
+int i2c_read_reg(ic2_t dev, uint16_t addr, uint16_t reg,
+                 void *data, size_t len, uint8_t flags);
+
+/**
+ * @brief   Write data from/to the given I2C device
+ *
+ * @param[in]  dev          I2C peripheral device
+ * @param[in]  addr         7-bit or 10-bit device address (right-aligned)
+ * @param[out] data         array holding the received bytes
+ * @param[in]  len          the number of bytes to read into @p data
+ * @param[in]  flags        optional flags (see @ref i2c_flags_t)
+ *
+ * @return                  I2C_ACK on successful transfer of @p len byte
+ * @return                  I2C_ADDR_NACK if response to address byte was NACK
+ * @return                  I2C_ERR for any other error
+ */
+int i2c_write(i2c_t dev, uint16_t addr,
+              const void *data, size_t len, uint8_t flags);
+
+/**
+ * @brief   Convenience function for writing a single byte onto the bus
```

Comment:
`for writing a single byte onto the bus` to `for writing a single byte to a device` ? or something like that? But I guess you also would like to use this for general call purposes?